internal/grpc/credentials.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+package grpc
+
+import "context"
+
+const (
+	TokenKey = "token"
+)
+
```

Comment:
This would be a great place to verify interface compliance: https://github.com/uber-go/guide/blob/master/style.md#verify-interface-compliance
```suggestion
const (
	TokenKey = "token"
)

var _ credentials.perRPCCredentials = (*PerRPCCredentials)(nil)
```

Reply:
nice! Added to next commit